### PR TITLE
feat: Resources support [DEV-1616]

### PR DESCRIPTION
**Pull request by askolesov**

### CODE REVIEW

**Review by Eengineer1 — APPROVED** [review]
@askolesov Cool encoding workaround & clean work.
Added more coverage as well.
Approved, thanks!

File: src/signer.ts [hunk 1]
```diff
@@ -172,6 +173,8 @@ export class CheqdSigningStargateClient extends SigningStargateClient {
 		})
 	}
 
+	// What does this do?
+	// This is completely wrong
```
[the remark this answers is not in the document]
Comment (askolesov) [reply src/signer.ts]: Yeah, sorry, didn't mean to publish this. Those were notes to myself.

File: src/types.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export enum MethodSpecificIdAlgo {
 }
 
 export type TSignerAlgo = {
+    // This is wrong: there can be more then one signer of a given type
```
[the remark this answers is not in the document]
Comment (askolesov) [reply src/types.ts]: And the same as well. Will resolve those comments. Again, sorry, those are just reminders for myself to ask.

File: src/signer.ts [hunk 3]
```diff
@@ -189,6 +192,7 @@ export class CheqdSigningStargateClient extends SigningStargateClient {
 		return this.didSigners
 	}
 
+	// Verification method id can be related to different DIDDoc. Also, for resources we don't have access to the corresponding DIDDoc.
```
[the remark this answers is not in the document]
Comment (askolesov) [reply src/signer.ts]: The same thing. We've already discussed it.

**Comment by askolesov** [comment]
@Eengineer1 Thanks for review. Have pushed really minor changes and will merge once workflows are passed.

**Comment by askolesov** [comment]
@Eengineer1 Thanks for review. Have pushed really minor changes and will merge once workflows are passed.

**Review by Eengineer1 — APPROVED** [review]
Approved minor additions & thanks!

**Pull request merged**